api/src/main/java/org/apache/iceberg/expressions/Expressions.java

```diff
@@ -109,16 +111,46 @@ public static Expression not(Expression child) {
     return new UnboundPredicate<>(Expression.Operation.STARTS_WITH, ref(name), value);
   }
 
+  public static <T> UnboundPredicate<T> in(String name, T value, T... values) {
+    return predicate(Operation.IN, name,
+        Stream.concat(Stream.of(value), Stream.of(values))
+            .map(Literals::from).collect(Collectors.toSet()));
+  }
+
+  public static <T> UnboundPredicate<T> notIn(String name, T value, T... values) {
```

Comment:
I'd rather have the signature of these factory methods use just `T...`. I see why you'd do this to ensure that there is at least one value, but this is the primary way to create predicates and this is difficult to use if you're converting from something else because when you have to separate the first element from a set of values.

So to make it simpler for callers, this should accept any `T... values`. If values is empty, then it should return `alwaysTrue` (or `alwaysFalse` for `in`), and return an UnboundPredicate for anything else.

Reply:
@rdblue 
Thanks for the comment, I can make it to be `T...` here.

I still think it may be better to throw exception instead of return `alwaysTrue` because that will force the method to return `Expression`. When users use it, they have to explicitly cast it to the concrete class, which makes it not-easy to use. 

Please let me know your thoughts. thanks.

Reply:
I'd rather return `Expression` because there are cases where the caller might not know what is in the set and simply pass through. Forcing the caller to check whether the set is empty instead of returning an equivalent wouldn't be as easy to use.

Reply:
Also, can you add a variant that accepts `Collection<T>`?

Reply:
Done.